### PR TITLE
Enable iOS Share Extension with TableView

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Make the iOS Share extension use a UITableView when displaying the channel list to help with memory optimization as it recycles the rows.

This PR Should be tested on an actual device in Release mode as the simulator the memory limits are not enforced.

### CODE REVIEW

**Comment by stephenkiers** [comment]
Seems to crash on my iphone
https://youtu.be/52m85RTCoAk

**Comment by enahum** [comment]
As the description says, run it in release mode

**Comment by stephenkiers** [comment]
Tested on iphone7plus. Worked great

File: share_extension/ios/extension_channels.js [hunk 1]
```diff
@@ -112,6 +111,11 @@ export default class ExtensionChannels extends PureComponent {
             const direct = getDirectChannels({entities});
             const channels = channelIds.map((id) => this.props.entities.channels.channels[id]).concat(direct);
 
+            this.publicChannelIcon = await Icon.getImageSource('globe', 16, this.props.theme.centerChannelColor);
```
Comment (hmhealey) [inline share_extension/ios/extension_channels.js]: Would it make any noticeable difference if we loaded these in parallel instead of one at a time?

Comment (enahum) [reply share_extension/ios/extension_channels.js]: Don’t know I can test it

**Comment by enahum** [comment]
@hmhealey changes made

**Pull request merged**